### PR TITLE
YuE2: support the safetensors-only v5/v8/v9 semantic heads, key latent cache by head

### PR DESCRIPTION
**Pull request by vdruts**
The YuE2 trainer tokenizes training audio with the community semantic head from `Mothersuperior/yue2-mothersuperior-realaudio-tokenizer-v4`. That repo now also carries the v5 / v8 / v9 joint pairs (same architecture; v8/v9 were co-trained with an audio-domain loss against the original recordings, and upstream prefers v9 by ear). They are published as **safetensors only**, and the current loaders only read the original `.pt` layout, so `model_kwargs.semantic_head_path` cannot point at them today.

This keeps v4 as the default and makes the newer files usable:

- **`src/tokenizer.py`**: `load_head_state_dict()` reads either the `.pt` (`{"model": ...}`) or a `.safetensors` head (the flat state dict).
- **`src/model.py`**: `merge_nar_lora` also accepts the named-tensor `.safetensors` NAR adapter and rebuilds the layer-major A/B list + `io` weights the merge already walks. The `_comfyui` fused layout is rejected with a clear error.
- **`yue2_model.py`**: the latent cache stores the head's codec tokens, so a non-default `semantic_head_path` now gets its own latent-space version (`..._tokenizer-head-joint-v9`). The default path keeps the existing key, so nobody's current cache is invalidated. Without this, switching heads silently reuses the old head's tokens.

Usage:

```yaml
model_kwargs:
  semantic_head_path: Mothersuperior/yue2-mothersuperior-realaudio-tokenizer-v4/tokenizer_head_joint_v9.safetensors
  # only if merge_nar_lora is on:
  nar_lora_path: Mothersuperior/yue2-mothersuperior-realaudio-tokenizer-v4/nar_lora_joint_v9.safetensors
```

### Tested (Windows, RTX 5090, torch 2.x)

- v4 head: `.safetensors` vs `.pt` state dicts are bit-exact through the new loader; v4 and v9 safetensors both `load_state_dict(strict=True)` into `TokenizerHead`.
- v4 NAR adapter: the list rebuilt from `.safetensors` is bit-exact with the `.pt` (`lora` 392 tensors + `vae2llm` / `llm2vae`); the v9 adapter loads with the same shapes; the `_comfyui` file raises.
- End to end through `SemanticTokenizer.tokenize` on three full songs: v4 and v9 agree on only 22-31% of frames, so the head choice really does change the training targets, which is why the cache key matters.
- Cache key: default -> unchanged, default + separation -> `_sep` as before, v9 -> `_tokenizer-head-joint-v9`, v9 + separation -> `_tokenizer-head-joint-v9_sep`.

Not tested: a full training run on v9 tokens, and the `merge_nar_lora` path on the bf16 checkpoint beyond the tensor-level equivalence above. I have not changed the default because I have no listening A/B yet showing v9 tokens train a better LoRA; happy to flip it (or add a UI dropdown) if you want that in this PR.

Related: #1052

### CODE REVIEW

**Comment by vdruts** [comment]
Follow-up: ran this branch through a real job to close the "not tested" gap on the training path.

- 15-song dataset that already had a v4 latent cache, same config as an earlier run, only `model_kwargs.semantic_head_path` pointed at a local `tokenizer_head_joint_v9.safetensors`.
- The cacher built 15 new files with `latent_space_version: yue2_tokenizer-head-joint-v9` in their metadata and left the 15 existing `yue2` files untouched, so both caches now coexist in `_latent_cache`. Spot check on one song: the old and new `dto.tokens` agree on about 25 % of frames.
- Training then ran the full 350 steps (int8 convrot, `cot: full`, ~3 s/it on a 5090) and saved every checkpoint normally.

Still not exercised: `merge_nar_lora` with the safetensors adapter on the bf16 checkpoint (only the tensor-level equivalence with the `.pt` is verified). Listening comparison of v4-head vs v9-head LoRAs is rendering now; I'll report if it says anything useful about which head should be the default.
